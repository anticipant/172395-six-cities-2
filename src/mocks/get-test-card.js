export const getTestCard = (name, number, bool) => ({
  id: number,
  city: {
    name,
    location: {
      latitude: number,
      longitude: number,
      zoom: number
    }
  },
  title: `Title-${name}`,
  isPremium: bool,
  previewImage: `img/apartment-${number}.jpg`,
  price: number,
  rating: number,
  type: `Apartment 3`,
  images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
  isFavorite: !bool,
  bedrooms: 3,
  maxAdults: 4,
  goods: [`Heating-${name}`, `Kitchen-${name}`, `Cable TV -${name}`, `Washing machine-${name}`, `Coffee machine-${name}`, `Dishwasher-${name}`],
  host: {
    id: 3,
    isPro: bool,
    name: `Angelina-${name}`,
    avatarUrl: `img/${number}.png`
  },
  description: `${name} A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: number,
    longitude: number,
    zoom: number
  },
});
