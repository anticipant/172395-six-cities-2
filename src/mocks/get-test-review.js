export const getTestReview = (name, number, bool) => ({
  id: number,
  user: {
    id: number + 1,
    isPro: bool,
    name,
    avatarUrl: `img/${name}.jpg`
  },
  rating: number,
  comment: `${name} A quiet cozy ${name}  and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`
});
