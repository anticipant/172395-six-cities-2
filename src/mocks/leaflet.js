export const leaflet = {
  icon: () => {},
  map: () => ({
    setView: () => {},
    addTo: () => {},
  }),
  tileLayer: () => ({
    addTo: () => {},
  }),
  marker: () => ({
    addTo: () => {},
  }),
};
