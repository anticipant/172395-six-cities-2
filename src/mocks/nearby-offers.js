export const nearbyOffers = [
  {
    id: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 10
      }
    },
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Nearby 1 Beautiful & luxurious studio at great location`,
    isFavorite: true,
    isPremium: true,
    rating: 5,
    type: `apartment`,
    bedrooms: 5,
    maxAdults: 5,
    price: 555,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 5,
      isPro: true,
      name: `Angelina 5`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.39095539468,
      longitude: 4.8530966646,
      zoom: 8
    },
  },
  {
    id: 3,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 10
      }
    },
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Nearby 3 Beautiful & luxurious studio at great location`,
    isFavorite: true,
    isPremium: true,
    rating: 3,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 3,
    price: 333,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina 3`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 51.390,
      longitude: 4.854,
      zoom: 8
    },
  },
  {
    id: 2,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 53.390,
        longitude: 4.870,
        zoom: 10
      }
    },
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Nearby 2 Beautiful & luxurious studio at great location`,
    isFavorite: true,
    isPremium: true,
    rating: 4,
    type: `apartment`,
    bedrooms: 4,
    maxAdults: 4,
    price: 444,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 4,
      isPro: true,
      name: `Angelina 4`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.420,
      longitude: 4.856,
      zoom: 8
    },
  },
];
