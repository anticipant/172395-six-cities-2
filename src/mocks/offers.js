export const offers = [
  {
    id: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 10
      }
    },
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: true,
    isPremium: true,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 8
    },
  },
  {
    id: 2,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 10
      }
    },
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Wood and stone place`,
    isFavorite: false,
    isPremium: true,
    rating: 20,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 2222,
    goods: [`Heating2`, `Kitchen1`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 2,
      isPro: true,
      name: `Angelina2`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A2 quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198,
      zoom: 8
    },
  },
  {
    id: 3,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 10
      }
    },
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Wood and stone place`,
    isFavorite: false,
    isPremium: false,
    rating: 33,
    type: `apartment`,
    bedrooms: 33,
    maxAdults: 3,
    price: 33,
    goods: [`Heating3`, `Kitchen1`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina3`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A3 quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198,
      zoom: 8
    },
  },
  {
    id: 4,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 10
      }
    },
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Wood and stone place`,
    isFavorite: false,
    isPremium: false,
    rating: 55,
    type: `apartment`,
    bedrooms: 55,
    maxAdults: 4,
    price: 44,
    goods: [`Heating3`, `Kitchen1`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 4,
      isPro: true,
      name: `Angelina4`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A4 quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.3809553943508,
      longitude: 4.939309666406198,
      zoom: 8
    },
  },
];
