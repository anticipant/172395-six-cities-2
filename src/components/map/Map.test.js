import React from 'react';
import renderer from 'react-test-renderer';
import Map from "./Map";
import {leaflet} from "../../mocks/leaflet";
import {getTestCard} from "../../mocks/get-test-card";

it(`Map correctly renders after relaunch`, () => {
  const offers = [getTestCard(`Map snapshot`, 888, true)];

  const tree = renderer.create(<Map name={`Map snapshot`} offers={offers} leaflet={leaflet}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
