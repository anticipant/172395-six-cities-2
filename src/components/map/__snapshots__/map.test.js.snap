// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map correctly renders after relaunch 1`] = `
<div
  id="map"
  style={
    Object {
      "height": undefined,
      "width": "100%",
    }
  }
/>
`;
